### ISSUE TITLE
Compare previous and new data by value and not reference

### ISSUE THREAD

**Issue by dominik-zeglen** [iddan/react-spreadsheet#234]
I'm wondering if it's done by design to compare by reference (`!==` operator), or it's been just overlooked?
I use `react-spreadsheet` in controlled mode and I ran into render loops because of reference mismatches.
 
Rel
https://github.com/iddan/react-spreadsheet/blob/d8b3fb167a35e3e38df3e33bb7d225e5c430a278/src/Spreadsheet.tsx#L216

**Comment by iddan** [comment]
Hey @dominik-zeglen, like all the React primitives values are compared by reference in Spreadsheet. The reference of `data` should only change when some of its values are changed otherwise you should preserve its reference. It's the same principle as [`useEffect()`](https://reactjs.org/docs/hooks-effect.html) for instance.

Can you please provide an example of where you ran into render loops so I can better understand your pain?

**Comment by dominik-zeglen** [comment]
Makes sense to me.
I have a bit complicated use case here, because I had to implement some functions like column & row reordering, data reloading etc. To achieve this I was using it as 100% controlled input, but after playing with this for a day or two I ended up storing `onChange` data within a ref and reloading data state only if columns, rows or initial data changed, much like `react-spreadsheet` internally works. So as much as I believe there is a bug somewhere (I managed somehow to make `react-spreadsheet` trigger onChange loop by itself), it could be me that was misusing library and should not be a problem for other users :) I'll close this issue for now and reopen it (I hope not) if I run into the same problem again and be able to provide some codesandbox for you to reproduce this.
